### PR TITLE
docs: add Filter Drawer docs

### PR DESCRIPTION
**Pull request by didoesdigital**
First draft docs for Filter Drawer component

### CODE REVIEW

File: site/docs/components/filter-drawer.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+The drawer can grow horizontally and vertically to comfortably fit a few filters and increasing numbers of applied filters. It can grow to a max width that matches the content area width.
+
+- Layout for filter fields:
+    - Use rows or columns to clearly define groups of filter components.
+- Layout for applied filters:
+    - As people apply filter values, the filter fields may stack, taking up more drawer space. Use analytics to answer general usage questions to inform optimal drawer layout. Consider internal scrolling behavior when the drawer grows too large.
+
+### Filter components
+
+Use the filter drawer to house [Select](/components/select) and [Date Picker](/components/date-picker) components. Do not include [Search Fields](/components/search-field), which always live outside of the filter drawer.
```
Comment (didoesdigital) [inline site/docs/components/filter-drawer.mdx:56]: Could be clearer. A multi-select may be searchable.

File: site/docs/components/filter-drawer.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+<WhenToUseAndWhenNotToUse>
+<WhenToUse>
+
+*   Use a filter drawer to house filtering on a page that applies to the content of the whole page.
+
+</WhenToUse>
+<WhenNotToUse>
+
+*   For filtering a specific section of a page, use a filter bar instead.
```
Comment (didoesdigital) [inline site/docs/components/filter-drawer.mdx:68]: Filter bar page will need to happen

Comment (ActuallyACat) [reply site/docs/components/filter-drawer.mdx:68]: I don't think it needs a page right away, something like: "A filter bar is a Card(? or container) that groups related filter actions, such as form inputs" would provide the placeholder definition until a page is created.

**Comment by didoesdigital** [comment]
This content is likely to be merged in https://github.com/cultureamp/kaizen-design-system/pull/590

**Comment by didoesdigital** [comment]
Merged in https://github.com/cultureamp/kaizen-design-system/pull/590

**Pull request closed without merging**